### PR TITLE
Qu auto/sorqa 399

### PR DESCRIPTION
**Pull request by lb-sgent**
image

### CODE REVIEW

**Review by jp-sgent — APPROVED** [review]

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CaseDirectorySteps.java [hunk 1]
```diff
@@ -456,6 +458,22 @@ public CaseDirectorySteps(
...
         });
 
+    Then(
+        "I check that created Case is visible with ([^\"]*) status",
+        (String vaccinationStatus) -> {
+          String caseUUID = apiState.getCreatedCase().getUuid();
+          Assert.assertTrue(
+              webDriverHelpers.isElementVisibleWithTimeout(getCaseResultsUuidLocator(caseUUID), 5),
+              "There is no case with expected status");
+          TimeUnit.SECONDS.sleep(2);
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CaseDirectorySteps.java:468]: do we need this sleep if we have the next wait condition?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java [hunk 2]
```diff
@@ -1116,6 +1119,41 @@ public ContactDirectorySteps(
               CONTACT_DIRECTORY_DETAILED_PAGE_APPLY_FILTER_BUTTON);
           webDriverHelpers.waitForPageLoadingSpinnerToDisappear(40);
         });
+
+    Then(
+        "I set vaccination contact status filter to ([^\"]*) and apply",
+        (String vaccinationStatus) -> {
+          webDriverHelpers.selectFromCombobox(
+              CONTACT_VACCINATION_STATUS_FILTER_COMBOBOX, vaccinationStatus);
+          TimeUnit.SECONDS.sleep(2);
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java:1128]: same, is sleep needed if the click method has the built in wait?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java [hunk 3]
```diff
@@ -1116,6 +1119,41 @@ public ContactDirectorySteps(
...
+
+          webDriverHelpers.clickOnWebElementBySelector(APPLY_FILTERS_BUTTON);
+        });
+
+    Then(
+        "I set contact vaccination status filter to ([^\"]*) and apply",
+        (String vaccinationStatus) -> {
+          webDriverHelpers.selectFromCombobox(
+              CONTACT_VACCINATION_STATUS_FILTER_COMBOBOX, vaccinationStatus);
+          webDriverHelpers.clickOnWebElementBySelector(APPLY_FILTERS_BUTTON);
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java:1138]: move out apply filters logic, so we can have a reusable method (make sure if it doesn't already exist)

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java [hunk 4]
```diff
@@ -1116,6 +1119,41 @@ public ContactDirectorySteps(
...
+          TimeUnit.SECONDS.sleep(15);
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java:1139]: why so long sleep? we have the wait for loading spinner to disappear if appears

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java [hunk 5]
```diff
@@ -1116,6 +1119,41 @@ public ContactDirectorySteps(
...
+        });
+
+    Then(
+        "I check that created Contact is visible with ([^\"]*) status",
+        (String vaccinationStatus) -> {
+          String contactUuid = apiState.getCreatedContact().getUuid();
+          Assert.assertTrue(
+              webDriverHelpers.isElementVisibleWithTimeout(getContactsByUUID(contactUuid), 5),
+              "There is no contact with expected uuid");
+          TimeUnit.SECONDS.sleep(2);
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java:1149]: again, why this sleep

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EditEventSteps.java [hunk 6]
```diff
@@ -699,6 +700,36 @@ public EditEventSteps(
...
+
+          person = collectPersonUuid();
+        });
+
+    Then(
+        "I set participant vaccination status to ([^\"]*)",
+        (String vaccinationStatus) -> {
+          webDriverHelpers.selectFromCombobox(VACCINATION_STATUS_COMBOBOX, vaccinationStatus);
+          webDriverHelpers.clickOnWebElementBySelector(SAVE_BUTTON);
+          TimeUnit.SECONDS.sleep(10);
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EditEventSteps.java:729]: is this sleep needed?

**Review by rdutu-vg — CHANGES_REQUESTED** [review]

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java (hunk 3 above)
Comment (lb-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java:1128]: @rdutu-vg 
Should I create method like webDriverHelpers.selectAndApplyFilter(field, value) or selectFilter(field, value) and applyFilter()?

File: sormas-e2e-tests/src/test/resources/features/sanity/web/CaseFilters.feature [hunk 7]
```diff
@@ -426,3 +426,25 @@ Feature: Case filter functionality
       | "Deceased"       | 0       | 1       | 0       | 0       |
       | "Recovered"      | 0       | 0       | 1       | 0       |
       | "Unknown"        | 0       | 0       | 0       | 1       |
+
+  @issue=SORQA-5969 @env_de
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/resources/features/sanity/web/CaseFilters.feature:430]: replace issue with tmsLink

File: sormas-e2e-tests/src/test/resources/features/sanity/web/ContactFilters.feature [hunk 8]
```diff
@@ -200,4 +200,28 @@ Feature: Contact filter functionality
     And I click "Nur Kontakte von anderen Instanzen" checkbox on Contact directory page for DE version
     And I click APPLY BUTTON in Contact Directory Page
     And I check that number of displayed contact results is 0
-    And I click "Nur Kontakte von anderen Instanzen" checkbox on Contact directory page for DE version
\ No newline at end of file
+    And I click "Nur Kontakte von anderen Instanzen" checkbox on Contact directory page for DE version
+
+  @issue=SORQA-5969 @env_de
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/resources/features/sanity/web/ContactFilters.feature:205]: replace issue with tmsLink

File: sormas-e2e-tests/src/test/resources/features/sanity/web/EventFilters.feature [hunk 9]
```diff
@@ -169,3 +169,21 @@ Feature: Event Directory filters check
     And I apply "Archived events" to combobox on Event Directory Page
     And I check the number of displayed Event results from All button is 0
 
+  @issue=SORQA-5969 @env_de
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/resources/features/sanity/web/EventFilters.feature:172]: replace issue with tmsLink

**Review by rdutu-vg — APPROVED** [review]

**Pull request merged**